src/main/groovy/edu/oregonstate/mist/cardsapi/db/CardDAO.groovy

```diff
@@ -54,6 +55,49 @@ interface CardDAO extends Closeable {
     @SqlQuery("SELECT CARD_INSTANCE_SEQ.NEXTVAL FROM DUAL")
     Integer getNextId()
 
+    @SqlUpdate ("""
+        UPDATE CARDS
+            SET TYPE_ID = (SELECT TYPE_ID FROM CARD_TYPES WHERE TYPE = :c.type),
+            NAME = :c.name,
+            COLOR_ID = (SELECT COLOR_ID FROM CARD_COLORS WHERE COLOR = :c.color),
+            RARITY_ID = (SELECT RARITY_ID FROM CARD_RARITIES WHERE RARITY = :c.rarity),
+            ENERGY = :c.energy,
+            DESCRIPTION = :c.description
+        WHERE CARDS.ID = :id
+        
+    """)
+    void putCard(@Bind("id") Integer id,
+                 @BindBean("c") Card card)
+
+    // Check if card exists
+    @SqlQuery ("""
+        SELECT COUNT(*)
+        FROM CARDS
+        WHERE CARDS.ID = :id
+    """)
+    Boolean cardExists(@Bind("id") Integer id)
+
+    @SqlQuery ("""
+        SELECT TYPE
+        FROM CARD_TYPES
+    """)
+    @RegisterMapper(ListsMapper)
+    List<String> getValidTypes()
```

Comment:
If you're just doing a single column query, you shouldn't need a custom mapper class to map it to a list of strings.